File: binding/src/libqmetatype.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qmetatype.html
pub const qmetatype = struct {

    /// New constructs a new QMetaType object.
    pub fn New(other: ?*anyopaque) QtC.QMetaType {
        return qtc.QMetaType_new(@ptrCast(other));
    }


    /// New2 constructs a new QMetaType object.
    pub fn New2(other: ?*anyopaque) QtC.QMetaType {
        return qtc.QMetaType_new2(@ptrCast(other));
    }


    /// New3 constructs a new QMetaType object.
    pub fn New3(typeVal: i32) QtC.QMetaType {
        return qtc.QMetaType_new3(typeVal);
    }


    /// New4 constructs a new QMetaType object.
    pub fn New4(d: ?*anyopaque) QtC.QMetaType {
        return qtc.QMetaType_new4(@ptrCast(d));
    }


    /// New5 constructs a new QMetaType object.
    pub fn New5() QtC.QMetaType {
        return qtc.QMetaType_new5();
    }


    /// New6 constructs a new QMetaType object.
    pub fn New6(param1: ?*anyopaque) QtC.QMetaType {
        return qtc.QMetaType_new6(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QMetaType_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QMetaType_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn RegisterNormalizedTypedef(normalizedTypeName: []u8, typeVal: QtC.QMetaType) void {
        const normalizedTypeName_str = qtc.libqt_string{
    .len = normalizedTypeName.len,
    .data = normalizedTypeName.ptr,
};
qtc.QMetaType_RegisterNormalizedTypedef(normalizedTypeName_str, @ptrCast(typeVal));
    }

    pub fn Type(typeName: []const u8) i32 {
        return qtc.QMetaType_Type(@ptrCast(typeName));
    }

    pub fn Type2(typeName: []u8) i32 {
        const typeName_str = qtc.libqt_string{
    .len = typeName.len,
    .data = typeName.ptr,
};
return qtc.QMetaType_Type2(typeName_str);
    }

    pub fn TypeName(typeVal: i32) []const u8 {
        const _ret = qtc.QMetaType_TypeName(typeVal);
return  std.mem.span(_ret);
    }

    pub fn SizeOf(typeVal: i32) i32 {
        return qtc.QMetaType_SizeOf(typeVal);
    }

    pub fn TypeFlags(typeVal: i32) i32 {
        return qtc.QMetaType_TypeFlags(typeVal);
    }

    pub fn MetaObjectForType(typeVal: i32) QtC.QMetaObject {
        return qtc.QMetaType_MetaObjectForType(typeVal);
    }

    pub fn Create(typeVal: i32) ?*anyopaque {
        return qtc.QMetaType_Create(typeVal);
    }

    pub fn Destroy(typeVal: i32, data: ?*anyopaque) void {
        qtc.QMetaType_Destroy(typeVal, @ptrCast(data));
    }

    pub fn Construct(typeVal: i32, where: ?*anyopaque, copyVal: ?*anyopaque) ?*anyopaque {
        return qtc.QMetaType_Construct(typeVal, @ptrCast(where), @ptrCast(copyVal));
    }

    pub fn Destruct(typeVal: i32, where: ?*anyopaque) void {
        qtc.QMetaType_Destruct(typeVal, @ptrCast(where));
    }

    pub fn IsRegistered(typeVal: i32) bool {
        return qtc.QMetaType_IsRegistered(typeVal);
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsValid(@ptrCast(self));
    }

    pub fn IsRegistered2(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsRegistered2(@ptrCast(self));
    }

    pub fn RegisterType(self: ?*anyopaque, ) void {
        qtc.QMetaType_RegisterType(@ptrCast(self));
    }

    pub fn Id(self: ?*anyopaque, ) i32 {
        return qtc.QMetaType_Id(@ptrCast(self));
    }

    pub fn SizeOf2(self: ?*anyopaque, ) i64 {
        return qtc.QMetaType_SizeOf2(@ptrCast(self));
    }

    pub fn AlignOf(self: ?*anyopaque, ) i64 {
        return qtc.QMetaType_AlignOf(@ptrCast(self));
    }

    pub fn Flags(self: ?*anyopaque, ) i32 {
        return qtc.QMetaType_Flags(@ptrCast(self));
    }

    pub fn Name(self: ?*anyopaque, ) []const u8 {
        const _ret = qtc.QMetaType_Name(@ptrCast(self));
return  std.mem.span(_ret);
    }

    pub fn Create2(self: ?*anyopaque, ) ?*anyopaque {
        return qtc.QMetaType_Create2(@ptrCast(self));
    }

    pub fn Destroy2(self: ?*anyopaque, data: ?*anyopaque) void {
        qtc.QMetaType_Destroy2(@ptrCast(self), @ptrCast(data));
    }

    pub fn Construct2(self: ?*anyopaque, where: ?*anyopaque) ?*anyopaque {
        return qtc.QMetaType_Construct2(@ptrCast(self), @ptrCast(where));
    }

    pub fn Destruct2(self: ?*anyopaque, data: ?*anyopaque) void {
        qtc.QMetaType_Destruct2(@ptrCast(self), @ptrCast(data));
    }

    pub fn Compare(self: ?*anyopaque, lhs: ?*anyopaque, rhs: ?*anyopaque) QtC.QPartialOrdering {
        return qtc.QMetaType_Compare(@ptrCast(self), @ptrCast(lhs), @ptrCast(rhs));
    }

    pub fn Equals(self: ?*anyopaque, lhs: ?*anyopaque, rhs: ?*anyopaque) bool {
        return qtc.QMetaType_Equals(@ptrCast(self), @ptrCast(lhs), @ptrCast(rhs));
    }

    pub fn IsDefaultConstructible(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsDefaultConstructible(@ptrCast(self));
    }

    pub fn IsCopyConstructible(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsCopyConstructible(@ptrCast(self));
    }

    pub fn IsMoveConstructible(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsMoveConstructible(@ptrCast(self));
    }

    pub fn IsDestructible(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsDestructible(@ptrCast(self));
    }

    pub fn IsEqualityComparable(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsEqualityComparable(@ptrCast(self));
    }

    pub fn IsOrdered(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_IsOrdered(@ptrCast(self));
    }

    pub fn Save(self: ?*anyopaque, stream: ?*anyopaque, data: ?*anyopaque) bool {
        return qtc.QMetaType_Save(@ptrCast(self), @ptrCast(stream), @ptrCast(data));
    }

    pub fn Load(self: ?*anyopaque, stream: ?*anyopaque, data: ?*anyopaque) bool {
        return qtc.QMetaType_Load(@ptrCast(self), @ptrCast(stream), @ptrCast(data));
    }

    pub fn HasRegisteredDataStreamOperators(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_HasRegisteredDataStreamOperators(@ptrCast(self));
    }

    pub fn Save2(stream: ?*anyopaque, typeVal: i32, data: ?*anyopaque) bool {
        return qtc.QMetaType_Save2(@ptrCast(stream), typeVal, @ptrCast(data));
    }

    pub fn Load2(stream: ?*anyopaque, typeVal: i32, data: ?*anyopaque) bool {
        return qtc.QMetaType_Load2(@ptrCast(stream), typeVal, @ptrCast(data));
    }

    pub fn UnderlyingType(self: ?*anyopaque, ) QtC.QMetaType {
        return qtc.QMetaType_UnderlyingType(@ptrCast(self));
    }

    pub fn FromName(name: []const u8) QtC.QMetaType {
        return qtc.QMetaType_FromName(@ptrCast(name));
    }

    pub fn DebugStream(self: ?*anyopaque, dbg: ?*anyopaque, rhs: ?*anyopaque) bool {
        return qtc.QMetaType_DebugStream(@ptrCast(self), @ptrCast(dbg), @ptrCast(rhs));
    }

    pub fn HasRegisteredDebugStreamOperator(self: ?*anyopaque, ) bool {
        return qtc.QMetaType_HasRegisteredDebugStreamOperator(@ptrCast(self));
    }

    pub fn DebugStream2(dbg: ?*anyopaque, rhs: ?*anyopaque, typeId: i32) bool {
        return qtc.QMetaType_DebugStream2(@ptrCast(dbg), @ptrCast(rhs), typeId);
    }

    pub fn HasRegisteredDebugStreamOperator2(typeId: i32) bool {
        return qtc.QMetaType_HasRegisteredDebugStreamOperator2(typeId);
    }

    pub fn Convert(fromType: QtC.QMetaType, from: ?*anyopaque, toType: QtC.QMetaType, to: ?*anyopaque) bool {
        return qtc.QMetaType_Convert(@ptrCast(fromType), @ptrCast(from), @ptrCast(toType), @ptrCast(to));
    }

    pub fn CanConvert(fromType: QtC.QMetaType, toType: QtC.QMetaType) bool {
        return qtc.QMetaType_CanConvert(@ptrCast(fromType), @ptrCast(toType));
    }

    pub fn View(fromType: QtC.QMetaType, from: ?*anyopaque, toType: QtC.QMetaType, to: ?*anyopaque) bool {
        return qtc.QMetaType_View(@ptrCast(fromType), @ptrCast(from), @ptrCast(toType), @ptrCast(to));
    }

    pub fn CanView(fromType: QtC.QMetaType, toType: QtC.QMetaType) bool {
        return qtc.QMetaType_CanView(@ptrCast(fromType), @ptrCast(toType));
    }

    pub fn Convert2(from: ?*anyopaque, fromTypeId: i32, to: ?*anyopaque, toTypeId: i32) bool {
        return qtc.QMetaType_Convert2(@ptrCast(from), fromTypeId, @ptrCast(to), toTypeId);
    }

    pub fn Compare2(lhs: ?*anyopaque, rhs: ?*anyopaque, typeId: i32, result: *i32) bool {
        return qtc.QMetaType_Compare2(@ptrCast(lhs), @ptrCast(rhs), typeId, @ptrCast(result));
    }

    pub fn Equals2(lhs: ?*anyopaque, rhs: ?*anyopaque, typeId: i32, result: *i32) bool {
        return qtc.QMetaType_Equals2(@ptrCast(lhs), @ptrCast(rhs), typeId, @ptrCast(result));
    }

    pub fn HasRegisteredConverterFunction(fromType: QtC.QMetaType, toType: QtC.QMetaType) bool {
        return qtc.QMetaType_HasRegisteredConverterFunction(@ptrCast(fromType), @ptrCast(toType));
    }

    pub fn HasRegisteredMutableViewFunction(fromType: QtC.QMetaType, toType: QtC.QMetaType) bool {
        return qtc.QMetaType_HasRegisteredMutableViewFunction(@ptrCast(fromType), @ptrCast(toType));
    }

    pub fn UnregisterConverterFunction(from: QtC.QMetaType, to: QtC.QMetaType) void {
        qtc.QMetaType_UnregisterConverterFunction(@ptrCast(from), @ptrCast(to));
    }

    pub fn UnregisterMutableViewFunction(from: QtC.QMetaType, to: QtC.QMetaType) void {
        qtc.QMetaType_UnregisterMutableViewFunction(@ptrCast(from), @ptrCast(to));
    }

    pub fn UnregisterMetaType(typeVal: QtC.QMetaType) void {
        qtc.QMetaType_UnregisterMetaType(@ptrCast(typeVal));
    }

    pub fn Iface(self: ?*anyopaque, ) QtC.QtPrivate__QMetaTypeInterface {
        return qtc.QMetaType_Iface(@ptrCast(self));
    }

    pub fn Iface2(self: ?*anyopaque, ) QtC.QtPrivate__QMetaTypeInterface {
        return qtc.QMetaType_Iface2(@ptrCast(self));
    }

    pub fn Create22(typeVal: i32, copyVal: ?*anyopaque) ?*anyopaque {
        return qtc.QMetaType_Create22(typeVal, @ptrCast(copyVal));
    }

    pub fn Id1(self: ?*anyopaque, param1: i32) i32 {
        return qtc.QMetaType_Id1(@ptrCast(self), param1);
    }

    pub fn Create1(self: ?*anyopaque, copyVal: ?*anyopaque) ?*anyopaque {
        return qtc.QMetaType_Create1(@ptrCast(self), @ptrCast(copyVal));
    }

    pub fn Construct22(self: ?*anyopaque, where: ?*anyopaque, copyVal: ?*anyopaque) ?*anyopaque {
        return qtc.QMetaType_Construct22(@ptrCast(self), @ptrCast(where), @ptrCast(copyVal));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QMetaType_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/metatype.html#types
pub const enums = struct {
    pub const Type = enum {
        pub const Bool: i32 = 1;
        pub const Int: i32 = 2;
        pub const UInt: i32 = 3;
        pub const LongLong: i32 = 4;
        pub const ULongLong: i32 = 5;
        pub const Double: i32 = 6;
        pub const Long: i32 = 32;
        pub const Short: i32 = 33;
        pub const Char: i32 = 34;
        pub const Char16: i32 = 56;
        pub const Char32: i32 = 57;
        pub const ULong: i32 = 35;
        pub const UShort: i32 = 36;
        pub const UChar: i32 = 37;
        pub const Float: i32 = 38;
        pub const SChar: i32 = 40;
        pub const Nullptr: i32 = 51;
        pub const QCborSimpleType: i32 = 52;
        pub const Void: i32 = 43;
        pub const VoidStar: i32 = 31;
        pub const QChar: i32 = 7;
        pub const QString: i32 = 10;
        pub const QByteArray: i32 = 12;
        pub const QBitArray: i32 = 13;
        pub const QDate: i32 = 14;
        pub const QTime: i32 = 15;
        pub const QDateTime: i32 = 16;
        pub const QUrl: i32 = 17;
        pub const QLocale: i32 = 18;
        pub const QRect: i32 = 19;
        pub const QRectF: i32 = 20;
        pub const QSize: i32 = 21;
        pub const QSizeF: i32 = 22;
        pub const QLine: i32 = 23;
        pub const QLineF: i32 = 24;
        pub const QPoint: i32 = 25;
        pub const QPointF: i32 = 26;
        pub const QEasingCurve: i32 = 29;
        pub const QUuid: i32 = 30;
        pub const QVariant: i32 = 41;
        pub const QRegularExpression: i32 = 44;
        pub const QJsonValue: i32 = 45;
        pub const QJsonObject: i32 = 46;
        pub const QJsonArray: i32 = 47;
        pub const QJsonDocument: i32 = 48;
        pub const QCborValue: i32 = 53;
        pub const QCborArray: i32 = 54;
        pub const QCborMap: i32 = 55;
        pub const Float16: i32 = 63;
        pub const QModelIndex: i32 = 42;
        pub const QPersistentModelIndex: i32 = 50;
        pub const QObjectStar: i32 = 39;
        pub const QVariantMap: i32 = 8;
        pub const QVariantList: i32 = 9;
        pub const QVariantHash: i32 = 28;
        pub const QVariantPair: i32 = 58;
        pub const QByteArrayList: i32 = 49;
        pub const QStringList: i32 = 11;
        pub const QFont: i32 = 4096;
        pub const QPixmap: i32 = 4097;
        pub const QBrush: i32 = 4098;
        pub const QColor: i32 = 4099;
        pub const QPalette: i32 = 4100;
        pub const QIcon: i32 = 4101;
        pub const QImage: i32 = 4102;
        pub const QPolygon: i32 = 4103;
        pub const QRegion: i32 = 4104;
        pub const QBitmap: i32 = 4105;
        pub const QCursor: i32 = 4106;
        pub const QKeySequence: i32 = 4107;
        pub const QPen: i32 = 4108;
        pub const QTextLength: i32 = 4109;
        pub const QTextFormat: i32 = 4110;
        pub const QTransform: i32 = 4112;
        pub const QMatrix4x4: i32 = 4113;
        pub const QVector2D: i32 = 4114;
        pub const QVector3D: i32 = 4115;
        pub const QVector4D: i32 = 4116;
        pub const QQuaternion: i32 = 4117;
        pub const QPolygonF: i32 = 4118;
        pub const QColorSpace: i32 = 4119;
        pub const QSizePolicy: i32 = 8192;
        pub const FirstCoreType: i32 = 1;
        pub const LastCoreType: i32 = 63;
        pub const FirstGuiType: i32 = 4096;
        pub const LastGuiType: i32 = 4119;
        pub const FirstWidgetsType: i32 = 8192;
        pub const LastWidgetsType: i32 = 8192;
        pub const HighestInternalId: i32 = 8192;
        pub const QReal: i32 = 6;
        pub const UnknownType: i32 = 0;
        pub const User: i32 = 65536;
    };

    pub const TypeFlag = enum {
        pub const NeedsConstruction: i32 = 1;
        pub const NeedsDestruction: i32 = 2;
        pub const RelocatableType: i32 = 4;
        pub const MovableType: i32 = 4;
        pub const PointerToQObject: i32 = 8;
        pub const IsEnumeration: i32 = 16;
        pub const SharedPointerToQObject: i32 = 32;
        pub const WeakPointerToQObject: i32 = 64;
        pub const TrackingPointerToQObject: i32 = 128;
        pub const IsUnsignedEnumeration: i32 = 256;
        pub const IsGadget: i32 = 512;
        pub const PointerToGadget: i32 = 1024;
        pub const IsPointer: i32 = 2048;
        pub const IsQmlList: i32 = 4096;
        pub const IsConst: i32 = 8192;
        pub const NeedsCopyConstruction: i32 = 16384;
        pub const NeedsMoveConstruction: i32 = 32768;
    };

};

File: binding/src/libqmetatype.cpp
#include <QByteArray>
#include <QByteArrayView>
#include <QChar>
#include <QDataStream>
#include <QDebug>
#include <QMetaObject>
#include <QMetaType>
#include <QPartialOrdering>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qmetatype.h>
#include "libqmetatype.h"
#include "libqmetatype.hxx"

QMetaType* QMetaType_new(const QMetaType* other) {
	 return new QMetaType(*other);
}

QMetaType* QMetaType_new2(QMetaType* other) {
	 return new QMetaType(*other);
}

QMetaType* QMetaType_new3(int typeVal) {
	 return new QMetaType(typeVal);
}

QMetaType* QMetaType_new4() {
	 return new QMetaType();
}

QMetaType* QMetaType_new5(const QMetaType* param1) {
	 return new QMetaType(*param1);
}

void QMetaType_CopyAssign(QMetaType* self, QMetaType* other) {
    *self = *other;
}

void QMetaType_MoveAssign(QMetaType* self, QMetaType* other) {
    *self = std::move(*other);
}

void QMetaType_RegisterNormalizedTypedef(const libqt_string normalizedTypeName, QMetaType* typeVal) {
	QMetaType::registerNormalizedTypedef(QByteArray(normalizedTypeName.data, normalizedTypeName.len), *typeVal);
}

int QMetaType_Type(const char* typeName) {
	return QMetaType::type(typeName);
}

int QMetaType_Type2(const libqt_string typeName) {
	return QMetaType::type(QByteArray(typeName.data, typeName.len));
}

const char* QMetaType_TypeName(int typeVal) {
	return QMetaType::typeName(typeVal);
}

int QMetaType_SizeOf(int typeVal) {
	return QMetaType::sizeOf(typeVal);
}

int QMetaType_TypeFlags(int typeVal) {
	return QMetaType::typeFlags(typeVal);
}

const QMetaObject* QMetaType_MetaObjectForType(int typeVal) {
	return QMetaType::metaObjectForType(typeVal);
}

void* QMetaType_Create(int typeVal) {
	return QMetaType::create(typeVal);
}

void QMetaType_Destroy(int typeVal, void* data) {
	QMetaType::destroy(typeVal, data);
}

void* QMetaType_Construct(int typeVal, void* where, const void* copyVal) {
	return QMetaType::construct(typeVal, where, copyVal);
}

void QMetaType_Destruct(int typeVal, void* where) {
	QMetaType::destruct(typeVal, where);
}

bool QMetaType_IsRegistered(int typeVal) {
	return QMetaType::isRegistered(typeVal);
}

bool QMetaType_IsValid(const QMetaType* self) {
	return self->isValid();
}

bool QMetaType_IsRegistered2(const QMetaType* self) {
	return self->isRegistered();
}

void QMetaType_RegisterType(const QMetaType* self) {
	self->registerType();
}

int QMetaType_Id(const QMetaType* self) {
	return self->id();
}

ptrdiff_t QMetaType_SizeOf2(const QMetaType* self) {
	return self->sizeOf();
}

ptrdiff_t QMetaType_AlignOf(const QMetaType* self) {
	return self->alignOf();
}

int QMetaType_Flags(const QMetaType* self) {
	return self->flags();
}

const char* QMetaType_Name(const QMetaType* self) {
	return self->name();
}

void* QMetaType_Create2(const QMetaType* self) {
	return self->create();
}

void QMetaType_Destroy2(const QMetaType* self, void* data) {
	self->destroy(data);
}

void* QMetaType_Construct2(const QMetaType* self, void* where) {
	return self->construct(where);
}

void QMetaType_Destruct2(const QMetaType* self, void* data) {
	self->destruct(data);
}

QPartialOrdering* QMetaType_Compare(const QMetaType* self, const void* lhs, const void* rhs) {
	return new QPartialOrdering(self->compare(lhs, rhs));
}

bool QMetaType_Equals(const QMetaType* self, const void* lhs, const void* rhs) {
	return self->equals(lhs, rhs);
}

bool QMetaType_IsDefaultConstructible(const QMetaType* self) {
	return self->isDefaultConstructible();
}

bool QMetaType_IsCopyConstructible(const QMetaType* self) {
	return self->isCopyConstructible();
}

bool QMetaType_IsMoveConstructible(const QMetaType* self) {
	return self->isMoveConstructible();
}

bool QMetaType_IsDestructible(const QMetaType* self) {
	return self->isDestructible();
}

bool QMetaType_IsEqualityComparable(const QMetaType* self) {
	return self->isEqualityComparable();
}

bool QMetaType_IsOrdered(const QMetaType* self) {
	return self->isOrdered();
}

bool QMetaType_Save(const QMetaType* self, QDataStream* stream, const void* data) {
	return self->save(*stream, data);
}

bool QMetaType_Load(const QMetaType* self, QDataStream* stream, void* data) {
	return self->load(*stream, data);
}

bool QMetaType_HasRegisteredDataStreamOperators(const QMetaType* self) {
	return self->hasRegisteredDataStreamOperators();
}

bool QMetaType_Save2(QDataStream* stream, int typeVal, const void* data) {
	return QMetaType::save(*stream, typeVal, data);
}

bool QMetaType_Load2(QDataStream* stream, int typeVal, void* data) {
	return QMetaType::load(*stream, typeVal, data);
}

QMetaType* QMetaType_UnderlyingType(const QMetaType* self) {
	return new QMetaType(self->underlyingType());
}

QMetaType* QMetaType_FromName(QByteArrayView* name) {
	return new QMetaType(QMetaType::fromName(*name));
}

bool QMetaType_DebugStream(QMetaType* self, QDebug* dbg, const void* rhs) {
	return self->debugStream(*dbg, rhs);
}

bool QMetaType_HasRegisteredDebugStreamOperator(const QMetaType* self) {
	return self->hasRegisteredDebugStreamOperator();
}

bool QMetaType_DebugStream2(QDebug* dbg, const void* rhs, int typeId) {
	return QMetaType::debugStream(*dbg, rhs, typeId);
}

bool QMetaType_HasRegisteredDebugStreamOperator2(int typeId) {
	return QMetaType::hasRegisteredDebugStreamOperator(typeId);
}

bool QMetaType_Convert(QMetaType* fromType, const void* from, QMetaType* toType, void* to) {
	return QMetaType::convert(*fromType, from, *toType, to);
}

bool QMetaType_CanConvert(QMetaType* fromType, QMetaType* toType) {
	return QMetaType::canConvert(*fromType, *toType);
}

bool QMetaType_View(QMetaType* fromType, void* from, QMetaType* toType, void* to) {
	return QMetaType::view(*fromType, from, *toType, to);
}

bool QMetaType_CanView(QMetaType* fromType, QMetaType* toType) {
	return QMetaType::canView(*fromType, *toType);
}

bool QMetaType_Convert2(const void* from, int fromTypeId, void* to, int toTypeId) {
	return QMetaType::convert(from, fromTypeId, to, toTypeId);
}

bool QMetaType_Compare2(const void* lhs, const void* rhs, int typeId, int* result) {
	return QMetaType::compare(lhs, rhs, typeId, result);
}

bool QMetaType_Equals2(const void* lhs, const void* rhs, int typeId, int* result) {
	return QMetaType::equals(lhs, rhs, typeId, result);
}

bool QMetaType_HasRegisteredConverterFunction(QMetaType* fromType, QMetaType* toType) {
	return QMetaType::hasRegisteredConverterFunction(*fromType, *toType);
}

bool QMetaType_HasRegisteredMutableViewFunction(QMetaType* fromType, QMetaType* toType) {
	return QMetaType::hasRegisteredMutableViewFunction(*fromType, *toType);
}

void QMetaType_UnregisterConverterFunction(QMetaType* from, QMetaType* to) {
	QMetaType::unregisterConverterFunction(*from, *to);
}

void QMetaType_UnregisterMutableViewFunction(QMetaType* from, QMetaType* to) {
	QMetaType::unregisterMutableViewFunction(*from, *to);
}

void QMetaType_UnregisterMetaType(QMetaType* typeVal) {
	QMetaType::unregisterMetaType(*typeVal);
}

void* QMetaType_Create22(int typeVal, const void* copyVal) {
	return QMetaType::create(typeVal, copyVal);
}

int QMetaType_Id1(const QMetaType* self, int param1) {
	return self->id(param1);
}

void* QMetaType_Create1(const QMetaType* self, const void* copyVal) {
	return self->create(copyVal);
}

void* QMetaType_Construct22(const QMetaType* self, void* where, const void* copyVal) {
	return self->construct(where, copyVal);
}

void QMetaType_Delete(QMetaType* self) {
    delete self;
}

